pkg/apis/psmdb/v1/psmdb_defaults.go

```diff
@@ -179,6 +182,60 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 	return nil
 }
 
+func (cr *PerconaServerMongoDB) EncryptionByVersion() *bool {
+	encryption := true
+	apiVersion := cr.APIVersion
+	if lastCR, ok := cr.Annotations["kubectl.kubernetes.io/last-applied-configuration"]; ok {
+		var newCR PerconaServerMongoDB
+		err := json.Unmarshal([]byte(lastCR), &newCR)
+		if err != nil {
+			encryption := false
+			return &encryption
+		}
+		apiVersion = newCR.APIVersion
+	}
+
+	versionSlice := strings.Split(strings.TrimLeft(apiVersion, "psmdb.percona.com/v"), "-")
+	checkVersion := []int{
+		0: 1,
+		1: 2,
+	}
+
+	for k, v := range versionSlice {
```

Comment:
could we use https://github.com/hashicorp/go-version or https://github.com/Masterminds/semver ?